release.py

```diff
@@ -17,7 +17,10 @@
 import requests
 
 
-JENKINS_URL = "https://jenkins.cryptography.io/job/cryptography-wheel-builder"
+JENKINS_URL = (
+    "https://ci.cryptography.io/job/cryptography-support-jobs/"
+    "job/wheel-builder/"
```

Comment:
Nit: remove trailing slash, so we don't produce duplicate slashes in our requests.